WeakAuras/RegionTypes/DynamicGroup.lua

```diff
@@ -1044,6 +1053,15 @@ local function modify(parent, region, data)
       controlPoint:SetShown(show)
       controlPoint:SetWidth(regionData.dimensions.width)
       controlPoint:SetHeight(regionData.dimensions.height)
+      if regionData.useAnchorPerUnit and regionData.anchorPerUnit == "UNITFRAME" then
+        local guid = UnitGUID(regionData.region.state.unit)
+        if guid then
+          WeakAuras.dyngroup_unitframe_monitor = WeakAuras.dyngroup_unitframe_monitor or {}
```

Comment:
Don't you need to ensure that the frame is removed from a potential previous registration? E.g. if the unit changed?